pages/instructorAssessmentManualGrading/instanceQuestion/instanceQuestion.js

```diff
@@ -66,7 +71,16 @@ router.post(
        * where they can edit score. Fundamentally, we need to rethink how to treat questions that are
        * manually graded within PrairieLearn and how to handle those score calculations.
        */
-      await sqlDb.callAsync('instance_questions_update_score', params);
+      const result = await sqlDb.callAsync('instance_questions_update_score', params);
+      if (result.rows[0]?.modified_since_query) {
+        const attempted_score = JSON.stringify({
+          score_perc: req.body.submission_score_percent,
+          feedback: req.body.submission_note,
+        });
+        return res.redirect(
+          req.baseUrl + `?attempted_score=${encodeURIComponent(attempted_score)}`
+        );
```

Comment:
@nwalters512 I'd appreciate your thoughts on this strategy. I'm using a GET query to send the attempted score/feedback information to the page, but I'm not sure if that's the best approach. Having the POST itself call the EJS renderer is an option too, but I'm not a fan of that approach. Any other thoughts?

Reply:
Alternative option: instead of the sproc just returning a boolean false, there could be the creation of a `grading_jobs` record, and the URL could have the attempted grading job ID instead. This would then cause the page to retrieve the score and feedback from the grading job.

Reply:
This is fine for now. Once we implement https://github.com/PrairieLearn/PrairieLearn/issues/5416, we can implement something like [Rails flash messages](https://api.rubyonrails.org/classes/ActionDispatch/Flash.html) to persist and fetch this information on the next request. Until then, query params will be our best option, I think. We do use this pattern elsewhere, FWIW.

Reply:
> I'm a bit concerned with using session-based info for this. If a grader chooses to discard their changes, I'm not sure if we should open it again if the grader decides to open the same submission, the persistent information may get weird. Even if this is expected behaviour, it should happen for the user, not for the session (i.e., if the grader switches devices or browsers, the persistent information should either go with them to the other device, or not show at all).

This information would persist only for a fraction of the second between the `POST` and the subsequent `GET`. If the subsequent `GET` fails or if the user manages to reload the page on a different device in the fraction of the second between `GET` and `POST`, I'm frankly perfectly comfortable with weird behavior here.

That said, I'm fine with attaching this information to a grading job instead 👍